examples/pdb2hpp.rs

```diff
@@ -611,22 +611,12 @@ fn write_class(filename: &str, class_name: &str) -> pdb::Result<()> {
     }
 
     // add all the needed types iteratively until we're done
-    loop {
-        // get the last element in needed_types without holding an immutable borrow
-        let last = match needed_types.iter().next_back() {
-            Some(n) => Some(*n),
-            None => None,
-        };
-
-        if let Some(type_index) = last {
-            // remove it
-            needed_types.remove(&type_index);
-
-            // add the type
-            data.add(&type_finder, type_index, &mut needed_types)?;
-        } else {
-            break;
-        }
+    while let Some(type_index) = needed_types.iter().next_back().copied() {
```

Comment:
hm, `next_back`. Can’t we just reverse the iterator?

Reply:
We cannot construct one reverse iterator because we need to both modify and use `needed_types` inside the loop.